{
  "Class": "vulnerability",
  "ID": "oval:com.redhat.cve:def:201912781",
  "Version": "631",
  "Metadata": {
    "Title": "CVE-2019-12781 Django: Incorrect HTTP detection with reverse-proxy connecting via HTTPS (moderate)",
    "References": [
      {
        "Source": "CVE",
        "RefID": "CVE-2019-12781",
        "RefURL": "https://access.redhat.com/security/cve/CVE-2019-12781"
      }
    ],
    "Description": "DOCUMENTATION: An HTTP detection flaw was discovered in Django.  If deployed behind a reverse-proxy connecting to Django via HTTPS,  django.http.HttpRequest.scheme() incorrectly detected client requests made using HTTP as using HTTPS.  This resulted in incorrect results for is_secure() and build_absolute_uri(), and HTTP requests were not correctly redirected to HTTPS in accordance with SECURE_SSL_REDIRECT.\n            STATEMENT: This issue does not affect any versions of python-django as shipped with Red Hat Update Infrastructure for Cloud Providers as the load balancer should not be configured to forward HTTP requests.",
    "Advisory": {
      "From": "secalert@redhat.com",
      "Severity": "Moderate",
      "Issued": {},
      "Updated": {}
    }
  },
  "Criteria": {
    "Operator": "OR",
    "Criterions": [
      {
        "Comment": "Red Hat Enterprise Linux must be installed",
        "TestRef": "oval:com.redhat.cve:tst:20137459006"
      }
    ],
    "Criterias": [
      {
        "Operator": "AND",
        "Criterions": [
          {
            "Comment": "Red Hat Enterprise Linux 7 is installed",
            "TestRef": "oval:com.redhat.cve:tst:20137459005"
          }
        ],
        "Criterias": [
          {
            "Operator": "OR",
            "Criterias": [
              {
                "Operator": "AND",
                "Criterions": [
                  {
                    "Comment": "python-django-bash-completion is installed",
                    "TestRef": "oval:com.redhat.cve:tst:201912308001"
                  },
                  {
                    "Comment": "python-django-bash-completion is signed with Red Hat redhatrelease2 key",
                    "TestRef": "oval:com.redhat.cve:tst:201912308002"
                  }
                ]
              },
              {
                "Operator": "AND",
                "Criterions": [
                  {
                    "Comment": "python2-django is installed",
                    "TestRef": "oval:com.redhat.cve:tst:201912308003"
                  },
                  {
                    "Comment": "python2-django is signed with Red Hat redhatrelease2 key",
                    "TestRef": "oval:com.redhat.cve:tst:201912308004"
                  }
                ]
              },
              {
                "Operator": "AND",
                "Criterions": [
                  {
                    "Comment": "python2-django-doc is installed",
                    "TestRef": "oval:com.redhat.cve:tst:201912308005"
                  },
                  {
                    "Comment": "python2-django-doc is signed with Red Hat redhatrelease2 key",
                    "TestRef": "oval:com.redhat.cve:tst:201912308006"
                  }
                ]
              },
              {
                "Operator": "AND",
                "Criterions": [
                  {
                    "Comment": "python-django is installed",
                    "TestRef": "oval:com.redhat.cve:tst:201912308007"
                  },
                  {
                    "Comment": "python-django is signed with Red Hat redhatrelease2 key",
                    "TestRef": "oval:com.redhat.cve:tst:201912308008"
                  }
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}